plugin/trino-opensearch/src/main/java/io/trino/plugin/opensearch/ScanQueryPageSource.java

```diff
@@ -158,8 +158,22 @@ public Page getNextPage()
             Map<String, Object> document = hit.getSourceAsMap();
 
             for (int i = 0; i < decoders.size(); i++) {
-                String field = columns.get(i).name();
-                decoders.get(i).decode(hit, () -> getField(document, field), columnBuilders[i]);
+                OpenSearchColumnHandle columnHandle = columns.get(i);
+                if (columnHandle.path().size() == 1) {
+                    decoders.get(i).decode(hit, () -> getField(document, columnHandle.path().getFirst()), columnBuilders[i]);
+                    continue;
+                }
+                Map<String, Object> value = (Map<String, Object>) getField(document, columnHandle.path().getFirst());
+                if (value != null) {
+                    for (int j = 1; j < columnHandle.path().size() - 1; j++) {
+                        value = (Map<String, Object>) getField(value, columnHandle.path().get(j));
+                        if (value == null) {
+                            break;
+                        }
+                    }
+                }
+                Map<String, Object> leafDocument = value;
```

Comment:
Move this to a separate method (e.g., `resolveField`). This has a few benefits:
* Avoids the nested unrelated loop with cryptic loop variables (`i` vs `j`)
* Makes it easier to follow what the code is doing by introducing an appropriate abstraction.
* Avoids having to create an artificial "effectively final" local variable to get around the limitations that disallow using `value` in the lambda expression below.

Reply:
Created a separate method.